common/script/index.js

```diff
@@ -1152,6 +1152,53 @@ api.wrap = function(user, main) {
             message: "+1 Gem"
           }, _.pick(user, $w('stats balance'))) : void 0;
         }
+/* buy 5 gems */
+        if (type === 'gems5' && key === 'gem') {
```

Comment:
The whole `purchase` function is terrible. We never should have lumped all these things together... gold -> gem purchases should have been their own separate route instead of being tacked onto `purchase`